tests/src/Unit/MappingTest.php

```diff
@@ -0,0 +1,187 @@
+            'image' => 'bundle-0',
+            'video' => 'bundle-1',
+          // Same bundle as node has, but different mapping.
+            'something' => 'bundle-2',
+          ],
+        ],
+      ],
+      'bundles are not mapped' => [
+        FALSE,
+        [
+          'node' => [
+            'article' => 'article',
+            'page' => 'page',
+          ],
+        ],
+      ],
+    ];
+  }
+
+  /**
+   * Tests that getBundleMapping() maps correctly.
+   *
+   * @param bool $enableMapping
+   *   Enable mapping.
+   * @param array $testData
+   *   The test data. A two dimensional array. Root keys are $entityTypeIds,
+   *   values are arrays of [<input bundle> => <expected mapped value>].
+   *
+   * @covers ::getBundleMapping
+   *
+   * @dataProvider providerGetBundleMapping
+   */
+  public function testGetBundleMapping(bool $enableMapping, array $testData) {
+    foreach ($testData as $entityTypeId => $bundleMappings) {
+      $mapping = new Mapping();
+      $mapping->enableMapping($enableMapping);
+
+      foreach ($bundleMappings as $value => $mappedValue) {
+        $this->assertSame($mappedValue, $mapping->getBundleMapping($entityTypeId, $value));
+        // Multiple invocations must return the same mappings.
+        $this->assertSame($mappedValue, $mapping->getBundleMapping($entityTypeId, $value));
+      }
+    }
+  }
+
+  /**
+   * Data provider for testGetUserRoleMapping().
+   *
+   * @see ::testGetUserRoleMapping()
+   *
+   * @return array
+   *   The test data.
+   */
+  public function providerGetUserRoleMapping() {
+    return [
+      'roles are mapped' => [
+        TRUE,
+        [
+          'editor' => 'role-0',
+          'administrator' => 'role-1',
+          'something' => 'role-2',
+        ],
+      ],
+      'roles are not mapped' => [
+        FALSE,
+        [
+          'editor' => 'editor',
+          'administrator' => 'administrator',
+          'something' => 'something',
+        ],
+      ],
+    ];
+  }
+
+  /**
+   * Tests that getUserRoleMapping() maps correctly.
+   *
+   * @param bool $enableMapping
+   *   Enable mapping.
+   * @param array $testData
+   *   The test data. An array with input roles as keys and mapped values as
+   *   values.
+   *
+   * @covers ::getUserRoleMapping
+   *
+   * @dataProvider providerGetUserRoleMapping
+   */
+  public function testGetUserRoleMapping(bool $enableMapping, array $testData) {
+    $mapping = new Mapping();
+    $mapping->enableMapping($enableMapping);
+
+    foreach ($testData as $role => $mappedValue) {
+      $this->assertSame($mappedValue, $mapping->getUserRoleMapping($role));
+      // Multiple invocations must return the same mappings.
+      $this->assertSame($mappedValue, $mapping->getUserRoleMapping($role));
+    }
+  }
+
+  /**
+   * Data provider for testGetFieldMapping().
+   *
+   * @see ::testGetFieldMapping()
+   *
+   * @return array
+   *   The test data.
+   */
+  public function providerGetFieldMapping() {
```

Comment:
Makes sense

Reply:
But isn't that covered with the second test anyway?

Reply:
It could be, but I'm not sure is the case when names are anonymized covered. For example, it could happen that with some changes we get anonymization on `entity_id + bundle_type` namespace for fields and that's what should be nice to have covered.